core/iwasm/libraries/libc-wasi/libc_wasi_wrapper.c

```diff
@@ -1027,7 +1027,18 @@ execute_interruptible_poll_oneoff(
             return err;
         }
 
-        if (wasm_cluster_is_thread_terminated(exec_env)) {
+#if WASM_ENABLE_THREAD_MGR != 0
+        WASMCluster *cluster = wasm_exec_env_get_cluster(exec_env);
+        if (cluster)
+            os_mutex_lock(&cluster->lock);
+#endif
+        bool is_thread_terminated = wasm_cluster_is_thread_terminated(exec_env);
```

Comment:
Suggest to lock `exec_env->wait_lock` in wasm_cluster_is_thread_terminated and keep this file unchanged.